community/datadog/templates/daemonset.yaml

```diff
@@ -0,0 +1,169 @@
+{{ toYaml .Values.agents.podLabels | indent 6 }}
+      {{- end }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: "{{ template "datadog.fullname" . }}"
+        {{- if .Values.agents.podLabels }}
+{{ toYaml .Values.agents.podLabels | indent 8 }}
+        {{- end }}
+      name: {{ template "datadog.fullname" . }}
+      annotations:
+        checksum/autoconf-config: {{ tpl (toYaml .Values.datadog.autoconf) . | sha256sum }}
+        checksum/confd-config: {{ tpl (toYaml .Values.datadog.confd) . | sha256sum }}
+        checksum/checksd-config: {{ tpl (toYaml .Values.datadog.checksd) . | sha256sum }}
+        {{- if .Values.agents.customAgentConfig }}
+        checksum/agent-config: {{ tpl (toYaml .Values.agents.customAgentConfig) . | sha256sum }}
+        {{- end }}
+        {{- if .Values.datadog.systemProbe.enabled }}
+        container.apparmor.security.beta.kubernetes.io/system-probe: {{ .Values.datadog.systemProbe.apparmor }}
+        container.seccomp.security.alpha.kubernetes.io/system-probe: {{ .Values.datadog.systemProbe.seccomp }}
+        {{- end }}
+        productID: {{ template "datadog.fullname" . }}
+        productName: {{ template "datadog.fullname" . }}
+        productVersion: "{{ .Chart.Version | replace "+" "_" }}"
+      {{- if .Values.agents.podAnnotations }}
+{{ toYaml .Values.agents.podAnnotations | indent 8 }}
+      {{- end }}
+    spec:
+      {{- if .Values.datadog.securityContext }}
+      securityContext:
+{{ toYaml .Values.datadog.securityContext | indent 8 }}
+      {{- end }}
+      {{- if .Values.agents.useHostNetwork }}
+      hostNetwork: {{ .Values.agents.useHostNetwork }}
+      dnsPolicy: ClusterFirstWithHostNet
+      {{- end }}
+      {{- if .Values.datadog.dogstatsd.useHostPID }}
+      hostPID: {{ .Values.datadog.dogstatsd.useHostPID }}
+      {{- end }}
+      {{- if .Values.agents.image.pullSecrets }}
+      imagePullSecrets:
+{{ toYaml .Values.agents.image.pullSecrets | indent 8 }}
+      {{- end }}
+      {{- if .Values.agents.priorityClassName }}
+      priorityClassName: {{ .Values.agents.priorityClassName }}
+      {{- end }}
+      containers:
+        {{- include "container-agent" . | nindent 8 }}
+        {{- if .Values.datadog.apm.enabled }}
+          {{- include "container-trace-agent" . | nindent 8 }}
+        {{- end }}
+        {{- if .Values.datadog.processAgent.enabled }}
+          {{- include "container-process-agent" . | nindent 8 }}
+        {{- end }}
+        {{- if .Values.datadog.systemProbe.enabled }}
+          {{- include "container-system-probe" . | nindent 8 }}
+        {{- end }}
+      initContainers:
+        {{ include "containers-init" . | nindent 8 }}
+        {{- if and .Values.datadog.systemProbe.enabled (eq .Values.datadog.systemProbe.seccomp "localhost/system-probe") }}
+          {{ include "system-probe-init" . | nindent 8 }}
+        {{- end }}
+      volumes:
+        - name: config
+          emptyDir: {}
+        - hostPath:
```

Comment:
Why is hostPath being used?

Reply:
The Datadog agent daemonset use hostPath for several reasons:
- logs aggregation
- access to host cgroup information
- CRI socket access